### ISSUE TITLE
Updates.checkForUpdatesAsync causes crash on ExpoGo

### ISSUE THREAD

**Issue by Joshandrews43** [expo/expo#12604]
### Summary

`Updates.checkForUpdateAsync()` is causing a crash on IOS in ExpoGo. I have not tested android. Also have not tested on standalone because I am scared of messing with my users. Also note I have the latest version of ExpoGo on the app store, I tried re-downloading it with no success.

If I comment out the function that checks for updates, I do not crash, I've tested & published ~20 builds at this point with various tests. There is no other code that could be causing the problem. 

Using v.0.4.2 of `expo-updates`.

Semi-related, I have over 5k instances of [this sentry issue](https://sentry.io/share/issue/16dab8e4015a4004bf2111c4308daade/) in my production app that is from the same line of code. It worked for over a year with no issues, but two times now it randomly starts crashing with no version changes. Back then I solved it with a `try / catch`, but now it seems to be a native crash and cannot be caught.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

40.0.0

### Environment

```
 Expo CLI 4.0.13 environment info:
    System:
      OS: macOS 11.2.3
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 12.18.1 - ~/.nvm/versions/node/v12.18.1/bin/node
      Yarn: 1.22.10 - /usr/local/bin/yarn
      npm: 6.14.5 - ~/.nvm/versions/node/v12.18.1/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.9.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.5977832
      Xcode: 12.4/12D4e - /usr/bin/xcodebuild
    npmPackages:
      expo: ^40.0.1 => 40.0.1 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: https://github.com/expo/react-native/archive/sdk-40.0.1.tar.gz => 0.63.2 
    npmGlobalPackages:
      expo-cli: 4.0.13
    Expo Workflow: managed
```

### Reproducible demo or steps to reproduce from a blank project

```
export default function App() {
  useEffect(() => {
    if (process.env.NODE_ENV !== 'development') {
      setTimeout(checkForUpdates, 5000);
    }
  }, []);

  const checkForUpdates = async () => {
    alert('checking for update'); // this gets called and shows up directly before crash
    Updates.checkForUpdateAsync() // causes crash here
      .then(async (update) => {
        alert('checked for update'); // never shows up 
      })
      .catch((e) => {
        alert('something went wrong') // also never shows up
      });
  };

  return (
       <Navigation />
  );
}
```

**Comment by brentvatne** [comment]
can you provide a minimal reproducible example? the code that you provided is not runnable. see https://stackoverflow.com/help/minimal-reproducible-example for more information. happy to investigate if you can do this

**Comment by Joshandrews43** [comment]
> can you provide a minimal reproducible example? the code that you provided is not runnable. see https://stackoverflow.com/help/minimal-reproducible-example for more information. happy to investigate if you can do this

Happy to if you could help me understand how I can do so? Since it doesn't work in dev, I'm not sure what I can give you other than an `expo init` typescript project with the above `App.tsx` as its only component (can render null, it would yield the same result).

**Comment by brentvatne** [comment]
@Joshandrews43 - if you're able to reproduce it in a new project using the blank code in a new project, plus a call to `Updates.checkForUpdateAsync()`, then that is sufficient. just tell us what steps to follow after cloning to see the crash

**Comment by Joshandrews43** [comment]
@brentvatne I was able to reproduce it on a fresh expo project (tabs config). 

The github is here: [https://github.com/Joshandrews43/expo-updates-issue-sample](https://github.com/Joshandrews43/expo-updates-issue-sample)

Steps to reproduce: 
1. Clone
2. publish an update with `expo publish --release-channel <channel-name>`
3. Open the app in ExpoGo

Tried this twice and it crashed both times.

**Comment by Joshandrews43** [comment]
@brentvatne So peculiar, after crashing the first three times, my sample project now works fine.

**Comment by Joshandrews43** [comment]
I can reproduce the exact same behavior every time if I set my `sdkVersion` in `app.json` to the wrong version before publishing. 

However, I have the correct `sdkVersion` set in my real project, so I'm not sure if that's any help.

**Comment by Joshandrews43** [comment]
This is still an issue (and it 100% was not my code), but I upgraded to expo 41 using `expo-upgrade` with zero other changes (and had no `sdkVersion` in my `app.json`) and its working now on my production application.

I believe it is using the wrong `sdkVersion` under the hood or something. I can't tell you what it was, but I've made 0 code changes other than upgrading to expo 41 and it works all of a sudden. 

I recommend the team look into it. Thanks for your help.

**Comment by brentvatne** [comment]
interesting! it's possible that this is a bug in sdk 40. if you end up being able to reproduce this in a minimal sdk 40 project we can investigate that

**Comment by jparkrr** [comment]
Here is a similar issue that has a reproducible example https://github.com/expo/expo/issues/12495

**Comment by brentvatne** [comment]
@jparkrr - please share a minimal runnable reproducible example

**Comment by esamelson** [comment]
Hey all -- thanks for the report, I've verified this crash. The affected methods are `Updates.checkForUpdateAsync` and `Updates.fetchUpdateAsync` in SDK 39 and 40 apps in version 2.19.1 of Expo Go for iOS. Note that **standalone apps built with `expo build:ios` are unaffected by this bug**, so you don't need to worry about users encountering this. These methods also work fine in SDK 38 and 41 apps in Expo Go.

We have a fix in the works but it may take a while for it to be deployed to the App Store version of Expo Go. In the meantime, you'll need to work around this by either avoiding use of these two methods in Expo Go for SDK 39/40 apps, upgrading such apps to SDK 41, or using the simulator build of Expo Go (for which we'll push out an update by the end of the day).

I'm really sorry for the inconvenience 😞 we're working to resolve this as quickly as we can!

**Comment by esamelson** [comment]
This has been resolved in Expo Go version 2.19.3, which is out in the App Store as of yesterday.

**Comment by mayying** [comment]
I am still seeing this issue now in Expo Go version 2.29.2
